src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -359,4 +360,95 @@ public void registerUserForCourse(User user, Course course) {
         auditEventRepository.add(auditEvent);
         log.info("User " + user.getLogin() + " has successfully registered for course " + course.getTitle());
     }
+
+    /**
+     * Get the active students for this particular course
+     *
+     * @param courseId the id of the course
+     * @param periodIndex an index indicating which time period, 0 is current week, -1 is one week in the past, -2 is two weeks in the past ...
+     */
+    public Integer[] getCourseStatistics(Long courseId, Integer periodIndex) {
+        Integer[] result = new Integer[12];
+        Arrays.fill(result, 0);
+        ZonedDateTime now = ZonedDateTime.now();
+
+        LocalDateTime localStartDate = now.toLocalDateTime().with(DayOfWeek.MONDAY);
+        LocalDateTime localEndDate = now.toLocalDateTime().with(DayOfWeek.SUNDAY);
+        ZoneId zone = now.getZone();
+        ZonedDateTime startDate = localStartDate.atZone(zone).minusWeeks(11 + (12 * (-periodIndex))).withHour(0).withMinute(0).withSecond(0).withNano(0);
+        ZonedDateTime endDate = periodIndex != 0 ? localEndDate.atZone(zone).minusWeeks(12 * (-periodIndex)).withHour(23).withMinute(59).withSecond(59)
+                : localEndDate.atZone(zone).withHour(23).withMinute(59).withSecond(59);
+        List<Map<String, Object>> outcome = courseRepository.getCourseStatistics(courseId, startDate, endDate);
+        List<Map<String, Object>> distinctOutcome = convertMapList(outcome, startDate);
+        return createResultArray(distinctOutcome, result, endDate);
+    }
+
+    /**
+     * the List of maps result contains duplicated entries. This method compares the values and returns a List<Map<String, Object>>
+     * without duplicated entries
+     *
+     * @param result the result given by the Repository call
+     * @param startDate the startDate of the period
+     * @return A List<Map<String, Object>> containing date and amount of active users in this period
+     */
+    private List<Map<String, Object>> convertMapList(List<Map<String, Object>> result, ZonedDateTime startDate) {
```

Comment:
If it filters out duplicated, why not renaming the method to `convertToDistinctMapList()` or something like `removeDuplicatesFromMapList()` ? :D

Reply:
The ways of the Lord are inscrutable :D done in c4f74cf093cfd8e2afd0455b877365689c3d8834